src/TestInternalGrains/StreamLifecycleTestGrains.cs

```diff
@@ -34,6 +34,54 @@ public StreamLifecycleTestGrainState()
         }
     }
 
+    public class GenericArg
+    {
+        public string A { get; private set; }
+        public int B { get; private set; }
+
+        public GenericArg(string a, int b)
+        {
+            A = a;
+            B = b;
+        }
+
+        public override bool Equals(object obj)
+        {
+            var item = obj as GenericArg;
+            if (item == null)
+            {
+                return false;
+            }
+
+            return A.Equals(item.A) && B.Equals(item.B);
+        }
+
+        public override int GetHashCode()
+        {
+            return B ^ A.GetHashCode();
```

Comment:
I'm very tired, so apologizes if this is false memory, but I have a faint recall of problems of this sort: http://stackoverflow.com/questions/892618/create-a-hashcode-of-two-numbers. Do they apply?